### PR TITLE
feat: Pcaexplorer pcaplot

### PR DESCRIPTION
**Pull request by tdayris**
### Description

This PR adds a wrapper for [pcaExplorer](https://www.bioconductor.org/packages/release/bioc/vignettes/pcaExplorer/inst/doc/pcaExplorer.html#pcaExplorer_User_Guide)'s `pcaplot` function.

### QC

* [X] I confirm that:

For all wrappers added by this PR, 

* there is a test case which covers any introduced changes,
* `input:` and `output:` file paths in the resulting rule can be changed arbitrarily,
* either the wrapper can only use a single core, or the example rule contains a `threads: x` statement with `x` being a reasonable default,
* rule names in the test case are in [snake_case](https://en.wikipedia.org/wiki/Snake_case) and somehow tell what the rule is about or match the tools purpose or name (e.g., `map_reads` for a step that maps reads),
* all `environment.yaml` specifications follow [the respective best practices](https://stackoverflow.com/a/64594513/2352071),
* wherever possible, command line arguments are inferred and set automatically (e.g. based on file extensions in `input:` or `output:`),
* all fields of the example rules in the `Snakefile`s and their entries are explained via comments (`input:`/`output:`/`params:` etc.),
* `stderr` and/or `stdout` are logged correctly (`log:`), depending on the wrapped tool,
* temporary files are either written to a unique hidden folder in the working directory, or (better) stored where the Python function `tempfile.gettempdir()` points to (see [here](https://docs.python.org/3/library/tempfile.html#tempfile.gettempdir); this also means that using any Python `tempfile` default behavior works),
* the `meta.yaml` contains a link to the documentation of the respective tool or command,
* `Snakefile`s pass the linting (`snakemake --lint`),
* `Snakefile`s are formatted with [snakefmt](https://github.com/snakemake/snakefmt),
* Python wrapper scripts are formatted with [black](https://black.readthedocs.io).
* Conda environments use a minimal amount of channels, in recommended ordering. E.g. for bioconda, use (conda-forge, bioconda, nodefaults, as conda-forge should have highest priority and defaults channels are usually not needed because most packages are in conda-forge nowadays).

## Summary by CodeRabbit

- **New Features**
	- Introduced environment specification files (`environment.linux-64.pin.txt` and `environment.yaml`) for easier setup of computational environments.
	- Added a `meta.yaml` configuration for the PCA-Explorer PCA-plot tool, enhancing visualization capabilities for PCA results.
	- Implemented a `Snakefile` for automating PCA plotting processes, improving reproducibility in data analysis workflows.
	- Launched a wrapper R script to streamline execution of PCA analysis within Snakemake workflows.
  
- **Tests**
	- Added a new test function to validate PCA plotting functionality within the testing suite.

### CODE REVIEW

**Review by johanneskoester — DISMISSED** [review]

File: bio/pcaexplorer/pcaplot/test/Snakefile [hunk 1]
```diff
@@ -0,0 +1,11 @@
+rule test_pcaexplorer_pcaplot:
+    input:
+        "dst.RDS",
+    output:
+        "pca.png",
```
[the remark this answers is not in the document]
Comment (tdayris) [reply bio/pcaexplorer/pcaplot/test/Snakefile]: No reason at all. The output is now SVG formatted.

File: bio/pcaexplorer/pcaplot/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+name: "PCA-Explorer PCA-plot"
+description: Plots the results of PCA on a 2-dimensional space
+url: https://www.bioconductor.org/packages/release/bioc/vignettes/pcaExplorer/inst/doc/pcaExplorer.html#91_pcaplot
+authors:
+  - Thibault Dayris
+input:
+  - dst: Path to DESeqTransform object, such as rlog or vst transformed data
```
Comment (fgvieira) [inline bio/pcaexplorer/pcaplot/meta.yaml:7]:
```suggestion
  - Path to DESeqTransform object, such as rlog or vst transformed data
```

File: bio/pcaexplorer/pcaplot/wrapper.R [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        )
+    }
+}
+
+cmd <- base::paste0("pcaExplorer::pcaplot(", extra, ")")
+base::message("Command line used: ")
+base::message(cmd)
+
+# Build output graph
+width <- 480
```
Comment (fgvieira) [inline bio/pcaexplorer/pcaplot/wrapper.R:42]: Would it make sense to have this as a `param`?

Comment (tdayris) [reply bio/pcaexplorer/pcaplot/wrapper.R:42]: It already was in params, but there was a (fixed) typo.

**Comment by tdayris** [comment]
I'll add the pin soon

**Review by fgvieira — APPROVED** [review]

**Pull request merged**